src/workflows/contrib/start_service.py

```diff
@@ -82,6 +82,32 @@ def run(
             help="Name of the service to start. Known services: "
             + ", ".join(known_services),
         )
+        parser.add_option(
+            "-l",
+            "--liveness",
+            dest="liveness",
+            action="store_true",
+            default=False,
+            help=(
+                "Expose a liveness check endpoint for this service."
+                "A return code of 200 indicates success."
+                "A return code of 408 indicates failure."
+            ),
+        )
+        parser.add_option(
+            "--liveness-port",
+            dest="liveness_port",
+            default=8000,
+            type="int",
+            help="Expose liveness check endpoint on this port.",
+        )
+        parser.add_option(
+            "--liveness-timeout",
+            dest="liveness_timeout",
+            default=30,
+            type="float",
+            help="Timeout for the liveness check (in seconds).",
+        )
```

Comment:
Does it make sense to have this explicitly controlled here, when the kubernetes manifest also defines a value for this via `timeoutSeconds`? If you wanted to e.g. make it longer than 30s, you would have to change it in two places... and having an instance where this would happen is possibly never something that you want to recover from.... not trying to catch this scenario also makes the code mildly simpler.

(this comes up because IIRC with prometheus somewhere they explicitly recommend for you not to implement your own timeout failures, but to allow it just to be configurable on the server-side)

Reply:
I agree, probably simpler just to remove the timeout here and rely on the Kubernetes `timeoutSeconds` parameter.